### DISCUSSION TITLE
Fzf appends `:0` or `:1` to all my entries

### DISCUSSION THREAD

**Discussion by litoj** [ibhagwan/fzf-lua#1167] [Q&A] [answered]
I create inputs to `fzf_exec` like this:
```lua
fzf_cb(table.concat({
  tostring(i),
  ansi.magenta(node:type()),
  ansi.green(tostring(start_row)),
  ansi.green(tostring(start_col)),
  lines[start_row]:gsub('^\t', indent),
}, ':'), function() coroutine.resume(co) end)
```

fzf shows the entries like:
```
> 50:function:640:1:  function previewer:new(o, opts, fzf_win):1
  46:function:614:7:  local function producer(fzf_cb):1
  45:function:606:0:function M.fzf_jump(targets, opts):0
```

it appends the `:1` at the end whenever the column number `!=0`, otherwise it appends `:0`

What can I do to get rid of this?

**Comment by ibhagwan** [comment]
This isn’t the entirety of the code, you’re calling something that adds the line at the end, `fzf_cb` by itself won’t add line number numbers, are you calling `make_entry.file`?

**Reply by litoj** [reply]
No, I thought I did, but I didn't end up actually using it.

The full code looks like this: https://github.com/JosefLitos/syntax-tree-surfer/blob/c8e71bd3f60fbf1ceb355efa3a6cf03f2ffce027/lua/syntax-tree-surfer/init.lua#L606-L682

**Reply by ibhagwan** [reply]
Oh I have a guess why this happens, you’re feeding `table.concat` directly into `fzf_cb`, `table.concat` probably returns a tuple, the concatenated string and a number, use `local entry = table.concat(…)` and then feed entry into `fzf_cb`, that way you guarantee only one value.

**Comment by ibhagwan** [comment, marked as answer]
I found your problem, you’re using `gsub` directly inside the table which does return a tuple (not `table.concat` as I previously assumed), both gsub return values get unpacked into the table and then used as the extra fields:

> Change this:
```lua
lines[start_row]:gsub('^\t', indent)
```

> To this:
```lua
-- This way we extract only the first tuple ret val
({ lines[start_row]:gsub('^\t', indent) })[1]
```

**Reply by litoj** [reply]
Oh, I totally forgot about that, thank you!

BTW your plugin is awesome (not to mention `fzf` itself). I used to use Telescope, but felt like getting further away from the cli, so I wanted to find something closer to the cli tools already available. This fits it perfectly and does everything I could want.

**Reply by ibhagwan** [reply]
Ty @JosefLitos ❤️
